processing/src/main/java/io/druid/query/DataSourceUtil.java

```diff
@@ -28,4 +33,22 @@ public static String getMetricName(DataSource dataSource)
     final List<String> names = dataSource.getNames();
     return names.size() == 1 ? names.get(0) : names.toString();
   }
+
+  public static String getMetricName(Iterable<DataSourceWithSegmentSpec> dataSources)
+  {
+    return StreamSupport.stream(dataSources.spliterator(), false)
+                 .map(DataSourceUtil::getMetricName)
+                 .collect(Collectors.joining(",", "[", "]"));
+  }
+
+  private static String getMetricName(DataSourceWithSegmentSpec spec)
+  {
+    final StringJoiner joiner = new StringJoiner("=", "[", "]");
```

Comment:
StringJoiner is overkill to concat key and value, it could be
```java
String value = ...
return "[" + getMetricName(spec.getDataSource()) + "=" + value + "]";
```

Also surrounding [] could be confusing (imply that the list could have more "elements"), e. g. Map.Entry's toString() is just `key=value`.